src/tools/perf/perftest.c

```diff
@@ -712,13 +770,25 @@ static ucs_status_t setup_mpi_rte(struct perftest_context *ctx)
     ucs_trace_func("");
 
     MPI_Comm_size(MPI_COMM_WORLD, &size);
-    if (size != 2) {
-        ucs_error("This test should run with exactly 2 processes (actual: %d)", size);
+
+    if ((ctx->params.super.flags & UCX_PERF_TEST_FLAG_LOOPBACK) &&
+        (size != 1)) {
+        ucs_error("This test should be run with 1 process "
+                  "in loopback case (actual: %d)", size);
+        return UCS_ERR_INVALID_PARAM;
+    }
+
+    if (!(ctx->params.super.flags & UCX_PERF_TEST_FLAG_LOOPBACK) &&
+        (size != 2)) {
+        ucs_error("This test should be run with exactly 2 processes "
+                  "in p2p case (actual: %d)", size);
         return UCS_ERR_INVALID_PARAM;
     }
 
     MPI_Comm_rank(MPI_COMM_WORLD, &rank);
-    if (rank == 1) {
+
+    /* Let to the last rank to print the results */
```

Comment:
This sentence is (grammatically) weird.
Maybe:
```
Let the last rank print the results
```
Or:
```
Allow the last rank to print the results
```
etc.

(Also in line 807)